src/components/app/utils/loader.js

```diff
@@ -67,7 +67,7 @@
             };
 
             this.moduleLoaded = function( type ) {
-                if ( type && this.pendingModules.hasOwnProperty( type ) ) {
+                if ( type && this.pendingModules.hasOwnProperty( type ) && this.pendingModules[ type ] > 0) {
```

Comment:
I think it would be better to prevent conditionally loaded modules from calling the `moduleLoaded` method. Otherwise the loaded promise could be resolved before all not conditionally loaded modules have been loaded.